src/AutoRest.CSharp/DataPlane/Generation/DataPlaneClientWriter.cs

```diff
@@ -142,7 +142,16 @@ private void WriteClientCtors(CodeWriter writer, DataPlaneClient client, BuildCo
             var clientOptionsName = ClientBuilder.GetClientPrefix(context.DefaultLibraryName, context);
             foreach (var scheme in schemes)
             {
-                if (scheme is AzureKeySecurityScheme azureKeySecurityScheme)
+                if (scheme is AzureKeySecurityScheme)
+                {
+                    throw new NotSupportedException($"{typeof(AzureKeySecurityScheme)} is not supported. Use {typeof(KeySecurityScheme)} instead");
+                }
+                if (scheme is AADTokenSecurityScheme)
+                {
+                    throw new NotSupportedException($"{typeof(AADTokenSecurityScheme)} is not supported. Use {typeof(OAuth2SecurityScheme)} instead");
+                }
```

Comment:
I wonder if you're throwing exception for AzureKey, AADToken credential, then why [this](https://github.com/Azure/autorest.csharp/blob/3bcb5037c5a078a6791b9f39a01901541c2a8069/test/TestProjects/PublicClientCtor/readme.md) test project is working fine?

Reply:
If I understand correctly, your question is: why PublicClientCtor has configuration `security: [AzureKey, AADToken]` but its credential schema still pass this check? 

If so, the answaer is: the M4 change does not change any configuration. If you check out the code model change it would be 
![image](https://user-images.githubusercontent.com/97225342/168508244-bcbd7059-1f70-4253-850c-3364b2968a9b.png)
So, the schema for PublicClientCtor is KeySecurityScheme and OAuth2SecurityScheme now.